sdk/dapp-kit/src/components/SuiClientProvider.tsx

```diff
@@ -2,29 +2,72 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { SuiClient, getFullnodeUrl } from '@mysten/sui.js/client';
-import { SuiClientContext } from '../hooks/useSuiClient.js';
-import { useMemo } from 'react';
+import type { SuiClientOptions } from '@mysten/sui.js/client';
+import { createContext, useMemo, useState } from 'react';
 
-export interface SuiClientProviderProps {
+type NetworkConfig = SuiClient | SuiClientOptions;
+type NetworkConfigs<T extends NetworkConfig = NetworkConfig> = Record<string, T>;
+
+export interface SuiClientProviderContext {
+	client: SuiClient;
+	queryKey: (key: unknown[]) => unknown[];
```

Comment:
mind elaborating on the intended usage of `queryKey`? I didn't see it used in any tests and was a little confused why the return signature is `unknown[]` vs `string[]`

Reply:
ended up just removing the queryKey thing and handling this in the query hook instead